internal/pkg/source/source.go

```diff
@@ -126,12 +127,23 @@ func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields field
 			default:
 				continue
 
-			// local variable or value returned from a call
-			case *ssa.Alloc, *ssa.Call:
-				if isProducedBySanitizer(v.(ssa.Value), conf) {
+			// Do not mark allocation values returned by sanitizers as new sources
+			case *ssa.Alloc:
+				if isProducedBySanitizer(v, conf) {
 					continue
 				}
 
+			// Do not mark values returned by sanitizers as new sources.
+			// Do mark as new sources values returned by field propagators.
+			case *ssa.Call:
+				if isProducedBySanitizer(v, conf) {
+					continue
+				}
+
+				if propagators.IsFieldPropagator(v) {
```

Comment:
Nit: consider adding a `continue` instruction at the end of this `if`. Indeed, if this call is identified as a field propagator, then it won't be identified as a source below. (For that reason,  the `continue` isn't actually necessary, but I believe it may help with readability.) It may be a good idea to add a comment explaining all of this.)

Reply:
Hmmmm... I think this loop control has gotten a little out of hand.  The fact that the switch needed a comment to explain its flow (which of course I failed to read) might be evidence enough of that.

What do you think of the this most recent commit?  Switch to flag if the instruction produces a source, and add the node if it does.  Short-circuiting is now handled by boolean evaluation rather than a `continue` jump.

Upon this closer inspection, I also noticed a comment which seems inaccurate...  See the TODO.

This is creeped past the PR intent, so I can revert all this and split it into a different PR if you prefer.

Reply:
I have replied on the other threads.

I think I would prefer the revert + split, thank you for suggesting that. I agree that this PR is doing a bit too much now.

Reply:
Good call.  This PR has been rolled back to the thing version, and the `continue` has been added per this thread's first comment.  To avoid too many merge conflicts, I'll open (separate) PRs for consolidating arguments and for this loop control after this PR is merged.